app/views/events/show.html.erb

```diff
@@ -74,11 +74,13 @@
     </div>
     <div class="tab-pane" id="event-availability-tab">
       <div id="event_availabilities" >
+        <% responses_and_assignments = ( @event.responses +
+                                Assignment.active.for_time_span(@event.start_time..@event.end_time)).to_a
+                              %>
+
         <%= render(:partial => "availabilities/availabilities_table",
-                   :locals => { :availabilities => ( @event.responses +
-                                Assignment.active.for_time_span(@event.start_time..@event.end_time)),
-                                :partial_availabilities => ( @event.partial_responses)} ) \
-                                 || "No Availabilities yet !" %>
+                   :locals => { :availabilities =>
+                                responses_and_assignments.presence || "No Availabilities yet!"})%>
```

Comment:
I think here, if `responses_and_assignments` is empty, you are passing a string but you need to pass an empty array.

Reply:
Also: We're doing a lot of variable definition & logic in this view, rather than in the controller(s).  Logic in controllers is easier to untangle (IMO), and definitely easier to test. Not a blocker, but a personal preference.  (Even in this kind of situation where the availabilities_table partial is going to be rendered by three different views.)